### DISCUSSION TITLE
How to confirm what extensions in riscv that supported by gem5 o3cpu?

### DISCUSSION THREAD

**Discussion by Ye-Jinhong** [gem5/gem5#999] [Q&A]
How to know exactly the extensions in riscv that gem5 O3CPU supporting? Whether there is a document describes that or I only to search the instrction in [decode.isa](https://github.com/gem5/gem5/blob/stable/src/arch/riscv/isa/decoder.isa)?

**Comment by BobbyRBruce** [comment]
On any RISCV CPU you can call `get_isa_string` which will return a string describing the ISA extensions included. You can consult https://five-embeddev.com/riscv-user-isa-manual/Priv-v1.12/naming.html for more information on deciphering this string.

The reason there's not one centralized piece of documentation describing each CPUs extensions is different CPUs can be setup to have, or not have, different extensions. It's a configurable parameter in the RISCV CPU SimObjects.
